adaptors_task/student/src/impl.cpp

```diff
@@ -1,8 +1,37 @@
 #include "impl.h"
-#include <vector>
+#include <map>
+#include <stack>
+#include <algorithm>
 
-const std::vector<std::pair<char, char>> brackets {
+const std::map<char, char> brackets {
     {'(', ')'},
     {'[', ']'},
     {'{', '}'}
 };
+
+bool isValid(const std::string &source)
+{
+    std::stack<char> checkBrackets;
+
+    auto isWrongPair = [](const std::map<char, char>& brackets, const char& symbol)
+    {
+        return std::any_of(brackets.begin(), brackets.end(),
+                                         [&symbol](const std::pair<const char, char>& pair) { return symbol == pair.second ? true : false; });
```

Comment:
Please avoid "true" and "false" here, you don't need them

Reply:
Done